pandas/_libs/tslibs/strptime.pyx

```diff
@@ -236,11 +236,11 @@ def array_strptime(
             if exact:
                 found = format_regex.match(val)
                 if not found:
-                    raise ValueError(f"time data \"{val}\" at position {i} doesn't "
+                    raise ValueError(f"time data \"{val}\" doesn't "
```

Comment:
Rather than trying to set `as position {i}` in each place where an error can occur, it's more reliable to just preprend `at position {i}` when an error is caught (see below)

Reply:
I haven't done the same in tslib.pyx because when `format` is provided, that code is going to be called here anyway in #50242
So for now I've just made sure the error messages match for the two paths